WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -34,15 +36,25 @@ public  async Task BuildPrivacySuggestionsAsync(Wallet wallet, TransactionInfo i
 	{
 		IsLoading = true;
 
+		Suggestions.Clear();
+		SelectedSuggestion = null;
+
+		if (!info.IsPrivate)
+		{
+			Suggestions.Add(new PocketSuggestionViewModel(SmartLabel.Merge(transaction.SpentCoins.Select(x => CoinHelpers.GetLabels(x)))));
+		}
+
 		var suggestions =
 			await ChangeAvoidanceSuggestionViewModel.GenerateSuggestionsAsync(info, wallet, transaction);
 
-		Suggestions.Clear();
-		SelectedSuggestion = null;
+		var hasChange = transaction.InnerWalletOutputs.Any(x => x.ScriptPubKey != info.Address.ScriptPubKey);
 
-		foreach (var suggestion in suggestions)
+		if (hasChange)
 		{
-			Suggestions.Add(suggestion);
+			foreach (var suggestion in suggestions.Where(x=>x.TransactionResult != transaction))
```

Comment:
`.Where(x=>x.TransactionResult != transaction)`

I don't this is necessary. `suggestions` does not contain the initial (not optimized) transaction.

Reply:
I know an edge case when you can make this happen.

Reply:
fixed.